### PR TITLE
Styled Dropdown

### PR DESCRIPTION
**Pull request by guilean**
**Checklist**
- [X] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://trello.com/c/mMYjbJyj

### CODE REVIEW

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  component: Dropdown,
+};
+
+export const Playground = () => (
+  <>
+    <h1>Playground</h1>
+    <Dropdown title={text('Title', 'Title')}>
+      {text('Content', 'Content')}
+    </Dropdown>
+  </>
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Dropdown/Dropdown.stories.tsx:16]: can we reproduce all cases we have?

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+describe('Dropdown', () => {
+  it('should render children when dropdown is open', () => {
+    render(
+      <Dropdown title="title">
+        <div>foo</div>
+      </Dropdown>
+    );
+    const expandMore = screen.getByTestId('ExpandMore');
+    fireEvent.click(expandMore);
+
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Dropdown/Dropdown.test.tsx:14]: remove empty line

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    <StyledWrapper ref={node} data-testid={testId}>
+      <StyledAnchor
+        className={className}
+        role="button"
+        onClick={() => setIsOpen((isOpen) => !isOpen)}
+      >
+        <Typography tag={Typography.Tags.Span} type={Typography.Types.Body}>
+          {title}
+        </Typography>
+        <StyledIcon testId={isOpen ? 'ExpandLess' : 'ExpandMore'}>
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx:47]: lowercase first letter please

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+} from './Dropdown.styled';
+
+export const Dropdown = ({
+  title,
+  className,
+  children,
+  testId = 'dropdown',
+}: Props) => {
+  const node = useRef(null);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx:22]:
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { Icon, Typography } from 'components/primitives';
+import {
+  expand_more as ExpandMore,
+  expand_less as ExpandLess,
+} from 'components/primitives/Icon/icons/navigation';
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx:6]:
```suggestion
} from 'components/primitives/Icon';
```

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx:22]: Done

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx (hunk 5 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx:6]: Done

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.tsx:47]: Done

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.test.tsx (hunk 2 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.test.tsx:14]: Done

File: packages/styled-components/src/components/primitives/Dropdown/Dropdown.stories.tsx (hunk 1 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.stories.tsx:16]: Added

Comment (tkanzakic) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.stories.tsx:25]: we have a list of items, items with icons and two columns right now

Comment (guilean) [reply packages/styled-components/src/components/primitives/Dropdown/Dropdown.stories.tsx:94]: Added

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**